classes/site/VersionCheck.inc.php

```diff
@@ -72,82 +72,55 @@ public static function getCurrentDBVersion()
-        }
-        if (isset($data['type'][0]['value'])) {
-            $versionInfo['type'] = $data['type'][0]['value'];
-        }
-        if (isset($data['release'][0]['value'])) {
-            $versionInfo['release'] = $data['release'][0]['value'];
-        }
-        if (isset($data['tag'][0]['value'])) {
-            $versionInfo['tag'] = $data['tag'][0]['value'];
-        }
-        if (isset($data['date'][0]['value'])) {
-            $versionInfo['date'] = $data['date'][0]['value'];
-        }
-        if (isset($data['info'][0]['value'])) {
-            $versionInfo['info'] = $data['info'][0]['value'];
-        }
-        if (isset($data['package'][0]['value'])) {
-            $versionInfo['package'] = $data['package'][0]['value'];
-        }
-        if (isset($data['patch'][0]['value'])) {
-            $versionInfo['patch'] = [];
-            foreach ($data['patch'] as $patch) {
-                $versionInfo['patch'][$patch['attributes']['from']] = $patch['value'];
+        $key = __METHOD__ . $path;
+        $cache = Cache::get($key);
+        if (!is_array($cache) || FileManager::isVirtualPath($path) || filemtime($path) > ($cache['createdAt'] ?? new DateTime())->getTimestamp()) {
+            $xml = new SimpleXMLElement(FileManager::getStream($path));
+            $version = [];
+            foreach (['application', 'class', 'type', 'release', 'tag', 'date', 'info', 'package', 'lazy-load', 'sitewide'] as $name) {
+                if (isset($xml->$name)) {
+                    $version[$name] = (string) $xml->$name;
+                }
             }
-        }
-        if (isset($data['class'][0]['value'])) {
-            $versionInfo['class'] = (string) $data['class'][0]['value'];
-        }
-
-        $versionInfo['lazy-load'] = (isset($data['lazy-load'][0]['value']) ? (int) $data['lazy-load'][0]['value'] : 0);
-        $versionInfo['sitewide'] = (isset($data['sitewide'][0]['value']) ? (int) $data['sitewide'][0]['value'] : 0);
-
-        if (isset($data['release'][0]['value']) && isset($data['application'][0]['value'])) {
-            $versionInfo['version'] = Version::fromString(
-                $data['release'][0]['value'],
-                $data['type'][0]['value'] ?? null,
-                $data['application'][0]['value'],
-                $data['class'][0]['value'] ?? '',
-                $versionInfo['lazy-load'],
-                $versionInfo['sitewide']
+            $version['sitewide'] = (int) ($version['sitewide'] ?? 0);
+            $version['lazy-load'] = (int) ($version['lazy-load'] ?? 0);
+            if (isset($xml->patch)) {
+                $version['patch'] = [];
+                foreach ($xml->patch as $patch) {
+                    $version['patch'][$patch['from']] = (string) $patch;
+                }
+            }
+            $cache = [
+                'createdAt' => new DateTime(),
+                'data' => $version
+            ];
+            Cache::put($key, $cache);
```

Comment:
Hmm, there should be an extra check here (`FileManager::isVirtualPath($path)`)

Reply:
(Don't forget this comment!)